.github/triage/jax_toolbox_triage/bisect.py

```diff
@@ -0,0 +1,122 @@
+import datetime
+import subprocess
+
+
+def get_commit_history(
+    worker,
+    package,
+    start,
+    end,
+    dir,
+    main_branch,
+    logger=None,
+    args=None,
+):
+    """
+    Get the commit history for a given package between two commits.
+
+    Args:
+        worker (Container): The container worker to execute commands.
+        package (str): The name of the package.
+        start (str): The starting commit hash.
+        end (str): The ending commit hash.
+        dir (str): The directory where the git repository is located.
+        main_branch (str): The main branch name. Defaults is the default branch of the repo.
+        logger (Logger, optional): Logger for debug information. Defaults to None.
+        args: Additional arguments that may contain cherry-pick commits.
+
+    Returns:
+        data: list, list of all the commits
+        cherry_pick_range: str, range of cherry pick commits if any
+    """
+    # In particular the end commit might not already be known if the older,
+    # passing, container is being used for triage.
+    commits_known = worker.exec(
+        [
+            "sh",
+            "-c",
+            f"git cat-file commit {start} && git cat-file commit {end}",
+        ],
+        policy="once_per_container",
+        workdir=dir,
+    )
+    if commits_known.returncode != 0:
+        if args.override_remotes and package in args.override_remotes:
+            remote_url = args.override_remotes.get(package, "origin")
+            fetch_cmd = ["git", "fetch", remote_url, start, end]
+            worker.check_exec(fetch_cmd, workdir=dir)
+        else:
+            # default behaviour
+            # re: https://stackoverflow.com/questions/4089430/how-to-determine-the-url-that-a-local-git-repository-was-originally-cloned-from
+            remote_url_result = worker.check_exec(
+                ["git", "config", "--get", "remote.origin.url"], workdir=dir
+            )
+            if remote_url_result.returncode == 0:
+                remote_url = remote_url_result.stdout().strip()
+                fetch_cmd = ["git", "fetch", remote_url, start, end]
+                worker.check_exec(fetch_cmd, workdir=dir)
+            else:
+                logger.error(
+                    "No remote 'origin' found and no override provided. Cannot fetch missing commits"
+                )
+                raise Exception("Cannot find commits and no remote is configured")
```

Comment:
As I suggested in https://github.com/NVIDIA/JAX-Toolbox/pull/1538#discussion_r2201181033 I think this can be made way shorter and more direct
```suggestion
        worker.check_exec(["git", "fetch", args.override_remotes.get(package, "origin"), start, end], workdir=dir)```